lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/serialization/NamingLookupRequestCodec.java

```diff
@@ -52,7 +52,7 @@ public NamingLookupRequestCodec(final IdentifierFactory factory) {
    */
   @Override
   public byte[] encode(final NamingLookupRequest obj) {
-    final List<CharSequence> ids = new ArrayList<>();
+    final List<String> ids = new ArrayList<>();
```

Comment:
This change seems odd, as `String` implements `CharSequence`. Also, the below calls `.toString()` explicitly which would be unneeded if we use `String` here.